engine/common/version/version_control.go

```diff
@@ -211,6 +215,8 @@ func (v *VersionControl) BlockFinalized(h *flow.Header) {
 // Returns expected errors:
 // - ErrOutOfRange if incoming block height is higher that last handled block height
 func (v *VersionControl) CompatibleAtBlock(height uint64) (bool, error) {
+	// TODO: needs check if height < finalized root (or set start = finalized root)
```

Comment:
we should return an `ErrOutOfRange` error for heights below the finalized root height.